examples/LidarISAM2Example.cpp

```diff
@@ -0,0 +1,518 @@
+/*
+In order for beginners to quickly understand how to use the iSAM2 algorithm, which is one of the better back-end optimization algorithms, 
+the factor graphs created in this code only contain odometry and loopback factors.
+It only needs to receive an odometry topic of type nav_msgs::Odometry with the name “/odometry” and then it will run
+*/
+
+#include <mutex>
+#include <thread>
+#include <nav_msgs/Path.h>
+#include <nav_msgs/Odometry.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/common/eigen.h>
+#include <pcl/kdtree/kdtree_flann.h>
+#include <ros/ros.h>
+#include <tf/transform_datatypes.h>
+#include <tf/transform_broadcaster.h>
+#include <eigen3/Eigen/Dense>
+
+#include <gtsam/geometry/Rot3.h>
+#include <gtsam/geometry/Pose3.h>
+#include <gtsam/slam/PriorFactor.h>
+#include <gtsam/slam/BetweenFactor.h>
+#include <gtsam/nonlinear/NonlinearFactorGraph.h>
+#include <gtsam/nonlinear/LevenbergMarquardtOptimizer.h>
+#include <gtsam/nonlinear/Marginals.h>
+#include <gtsam/nonlinear/Values.h>
+#include <gtsam/nonlinear/ISAM2.h>
+#include <gtsam/linear/NoiseModel.h>
+
+using namespace gtsam;
```

Comment:
I like you're using gtsam:: everywhere, especially as there are also pcl things. So try to just remove this?

Reply:
Thank you for your reply! It's an oversight on my part, there's really no need to add namespace here！